### ISSUE TITLE
Tutorial is outdated

### ISSUE THREAD

**Issue by KeShih** [GeomScale/dingo#107]
### Python Version
Google Colab currently uses `Python 3.10`, while the `dingo` project requires `Python 3.8`, making it difficult to install `dingo` on Google Colab.

### Outdated content
1. **Dependence and installation**:
https://github.com/GeomScale/dingo/blob/e89128c639066159c46fac03811925e166b82c18/tutorials/dingo_tutorial.ipynb#L462-L467
https://github.com/GeomScale/dingo/blob/e89128c639066159c46fac03811925e166b82c18/tutorials/dingo_tutorial.ipynb#L573
The README recommends using `poetry`.
https://github.com/GeomScale/dingo/blob/e89128c639066159c46fac03811925e166b82c18/README.md?plain=1#L60-L62

2. **Ignored lpsolve Library**:
https://github.com/GeomScale/dingo/blob/e89128c639066159c46fac03811925e166b82c18/README.md?plain=1#L44-L49

**Comment by advait-zx** [comment]
This is my first contribution i want to solve this issue what should i do to make impact

**Comment by dikshant182004** [comment]
hi @KeShih can i solve this issue

**Comment by KeShih** [comment]
> hi @KeShih can i solve this issue

I think we should abandon Google Colab and switch to something more convenient and ready-to-use. GitHub Codespaces might be a good choice.

Or package it into a Docker image.

**Comment by vissarion** [comment]
> I think we should abandon Google Colab and switch to something more convenient and ready-to-use. GitHub Codespaces might be a good choice.

@KeShih interesting, I haven't use Github Codespaces. What are the advantages over google colab? For example, how is this issue solved easier on codespaces vs colab?
 
> Or package it into a Docker image.

This should be independent from the cloud option and should be useful anyways, see https://github.com/GeomScale/dingo/pull/79 which is related, but the PR is not in good shape yet.

**Comment by KeShih** [comment]
> @KeShih interesting, I haven't use Github Codespaces. What are the advantages over google colab? For example, how is this issue solved easier on codespaces vs colab?

@vissarion GitHub Codespaces is also ready-to-use, providing a browser-based IDE (VS Code + Ubuntu). Switching Python versions and installing Poetry are hard in Google Colab, but possible in Codespaces because it offers a full Linux environment.

**Comment by vissarion** [comment]
OK, thanks, then it may worth it to have an alternative notebook in Github Codespaces and decide how we proceed and which notebook to keep.

**Comment by hariszaf** [comment]
I would suggest to have a version of [notebook](https://github.com/GeomScale/dingo/blob/develop/tutorials/dingo_tutorial.ipynb) able to run on a codespace.

**Comment by vissarion** [comment]
> I would suggest to have a version of [notebook](https://github.com/GeomScale/dingo/blob/develop/tutorials/dingo_tutorial.ipynb) able to run on a codespace.

Yes, this is what I mean but alternative. Would the same file `dingo_tutorial.ipynb` just run on codespace if we upload it?

**Comment by saksham242007** [comment]
Hi, I’d like to work on updating the outdated tutorial to match the current dingo API.
I’ll focus on fixing broken code snippets and instructions so the tutorial runs correctly with the latest version.
